src/inference/src/compilation_context.cpp

```diff
@@ -65,11 +65,43 @@ std::string NetworkCompilationContext::calculateFileInfo(const std::string& file
     return std::to_string(seed);
 }
 
-std::string NetworkCompilationContext::computeHash(CNNNetwork& network,
-                                                   const std::map<std::string, std::string>& compileOptions) {
-    OV_ITT_SCOPE(FIRST_INFERENCE, itt::domains::IE_LT, "NetworkCompilationContext::computeHash - CNN");
+std::string NetworkCompilationContext::compute_hash(const std::shared_ptr<ov::Model>& model,
+                                                    const std::map<std::string, std::string>& compileOptions) {
+    OV_ITT_SCOPE(FIRST_INFERENCE, itt::domains::IE_LT, "NetworkCompilationContext::compute_hash - Model");
 
-    IE_ASSERT(network.getFunction());
+    OPENVINO_ASSERT(model);
+
+    uint64_t seed = 0;
+    // 1. Calculate hash on function
+    ov::pass::Manager m;
+    m.register_pass<ngraph::pass::FixRtInfo>();
+    m.register_pass<ov::pass::Hash>(seed);
+    m.run_passes(model);
+
+    // 2. Compute hash on serialized data and options
+    for (const auto& kvp : compileOptions) {
+        seed = hash_combine(seed, kvp.first + kvp.second);
+    }
+
+    // 3. Add runtime information which may not be serialized
+    for (const auto& op : model->get_ordered_ops()) {
+        const auto& rt = op->get_rt_info();
+        for (const auto& rtMapData : rt) {
+            seed = hash_combine(seed, rtMapData.first);
+            std::stringstream strm;
+            rtMapData.second.print(strm);
+            seed = hash_combine(seed, strm.str());
+        }
+    }
+
+    return std::to_string(seed);
+}
```

Comment:
can we reuse common code? E.g. inside `compute_hash` for `CNNNetwork` we can:
1. Can `compute_hash` for `ov::Model`
2. Then compute extra hash information based on `CNNNetwork`s input and output info.